### DISCUSSION TITLE
Using Search criteria in Service.List call

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6670] [Help] [answered]
I have the below code with one search criteria that works alright .
How do I include another search criteria which is actually a formular of the  difference between the values of two fields in the table

**Working code**

`SmartERP.Purchases.PurchaseDetailsService.List({ Criteria: [Serenity.Criteria("PurchaseId"), '=', purchaseId] }, resp => {`
                       ` resp.Entities.forEach(element => {`
                          `        ......`
                       `  })`
                   `})`

**What I want is this**, but I dont know how to construct the syntax

`SmartERP.Purchases.PurchaseDetailsService.List({ Criteria: [Serenity.Criteria("PurchaseId"), '=', purchaseId] and [Serenity.Criteria("Quantity" - "QuantityReceived"), '>', 0] }, resp => {`
                       ` resp.Entities.forEach(element => {`
                          `        ......`
                       `  })`
                   `})`

**Comment by edwardch** [comment, marked as answer]
Not sure you can do the formula client side like that in a list request. 

Maybe try add the difference as a new field in your `PurchaseDetailsRow.cs`

```cs
[DisplayName("Quantity Difference")]
[Expression("(T0.[Quantity] - T0.[QuantityReceived])")]
public decimal? QuantityDifference
{
    get => fields.QuantityDifference[this];
    set => fields.QuantityDifference[this] = value;
}
```

And update your PurchaseDetails Grid/Dialog code like so

```ts
// add above the class
const flds = ClientRow.Fields;

// ...

var criteria: any[];
SmartERP.Purchases.PurchaseDetailsService.List({
    Criteria:
        Serenity.Criteria.and(
            criteria,
            [[flds.PurchaseId], Serenity.Criteria.Operator.eq, purchaseId],
            [[flds.QuantityDifference], Serenity.Criteria.Operator.gt, 0]
        )
}, response => {
    response.Entities.forEach(element => {
        // do stuff
    })
})
```
